### PR TITLE
Disable auto-detection of Kubeflow environment

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Several users have reported that they needed to unset the KUBERNETES_PORT environment variable in order to run multi-GPU training (#5254 and #16236). This is because the Trainer has auto-selected the Kubeflow environment despite the fact that the process wasn't launched using PyTorchJob. So the detection is unreliable.

I propose to disable the auto-detection. PyTorchJob users can still use the environment plugin by passing it to the trainer:

```py
from lightning.pytorch.plugins.environments import KubeflowEnvironment

trainer = Trainer(plugins=KubeflowEnvironment())
```

Closes https://github.com/Lightning-AI/lightning/issues/16236

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @justusschock @carmocca @awaelchli

### CODE REVIEW

File: src/lightning/fabric/plugins/environments/kubeflow.py [hunk 1]
```diff
@@ -41,12 +44,8 @@ def main_port(self) -> int:
...
     @staticmethod
     def detect() -> bool:
-        """Returns ``True`` if the current process was launched using Kubeflow PyTorchJob."""
-        required_env_vars = {"KUBERNETES_PORT", "MASTER_ADDR", "MASTER_PORT", "WORLD_SIZE", "RANK"}
-        # torchelastic sets these. Make sure we're not in torchelastic
-        excluded_env_vars = {"GROUP_RANK", "LOCAL_RANK", "LOCAL_WORLD_SIZE"}
-        env_vars = os.environ.keys()
-        return required_env_vars.issubset(env_vars) and excluded_env_vars.isdisjoint(env_vars)
+        # The Kubeflow environment can't be detected automatically
+        return False
```
Comment (awaelchli) [inline src/lightning/fabric/plugins/environments/kubeflow.py:48]: We could also raise an exception here instead of returning False.

**Review by carmocca — APPROVED** [review]
This is a breaking change, no?

**Comment by awaelchli** [comment]
One could label it as such yes. We don't have any usage guides for it however.

File: src/lightning/fabric/plugins/environments/kubeflow.py (hunk 1 above)
Comment (Borda) [reply src/lightning/fabric/plugins/environments/kubeflow.py:48]: Exception would be nicer...

Comment (awaelchli) [reply src/lightning/fabric/plugins/environments/kubeflow.py]: done :)

**Review by Borda — APPROVED** [review]
